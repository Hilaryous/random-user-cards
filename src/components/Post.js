// @flow
import React from 'react'
import Text from './Text'
import Header from './Header'
import type { PostType } from '../types/Post'

type Props = PostType

const Post = ({
  title,
  body,
}: Props) => (
  <div>
    <Header>
      {title}
    </Header>
    <Text modifier="post">
      {body}
    </Text>
  </div>
)

export default Post
