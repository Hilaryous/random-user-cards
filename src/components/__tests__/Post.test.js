/* eslint-disable function-paren-newline */
import React from 'react'
import renderer from 'react-test-renderer'
import Post from '../Post'

describe('Post', () => {
  it('shows the correct snapshot tree for the Post component', () => {
    const tree = renderer.create(
      <Post
        title="foo bar"
        body="This is the body"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
