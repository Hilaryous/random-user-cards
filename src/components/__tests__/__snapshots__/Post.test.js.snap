// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post shows the correct snapshot tree for the Post component 1`] = `
<div>
  <h1
    className="css-1ce92xy"
  >
    foo bar
  </h1>
  <p
    className="css-1itr8po"
  >
    This is the body
  </p>
</div>
`;
